php_phongo.c

```diff
@@ -953,6 +952,13 @@ static bool php_phongo_apply_options_to_uri(mongoc_uri_t *uri, bson_t *options T
 			continue;
 		}
 
+        if (!strcasecmp(key, MONGOC_URI_READPREFERENCE)) {
```

Comment:
I think this logic belongs in [`php_phongo_apply_rp_options_to_uri()`](https://github.com/mongodb/mongo-php-driver/blob/dd43b08320b35fa3fa3326a8093495dac066ffd2/php_phongo.c#L1113) (see linked position), which is the only place we currently read this option value.

When I suggested to "focus on `php_phongo_apply_options_to_uri()` (and similar functions)" in https://github.com/mongodb/mongo-php-driver/pull/601#discussion_r123811088, I did not mean to suggest we validate all options in `php_phongo_apply_options_to_uri()`. The code example I shared in that comment (for `mongoc_uri_option_is_int32(key)`) can be adapted for the read preference.

Essentially:

```
if (bson_iter_init_find_case(&iter, options, MONGOC_URI_READPREFERENCE) && BSON_ITER_HOLDS_UTF8(&iter)) {
    const char *str = bson_iter_utf8(&iter, NULL);
    /* ... */
}
```

Becomes:

```
if (bson_iter_init_find_case(&iter, options, MONGOC_URI_READPREFERENCE)) {
	if (BSON_ITER_HOLDS_UTF8(&iter)) {
		const char *str = bson_iter_utf8(&iter, NULL);
		/* ... */
	} else {
		/* phongo_throw_exception() for invalid type */
		return false;
	}
}
```

Notice that we do have to adapt the code a bit from my example in https://github.com/mongodb/mongo-php-driver/pull/601#discussion_r123811088 because `const char *str` must be declared at the top of its block scope. Therefore, we can't simply precede it with an `if` that conditionally returns.